docs/data-migration.md

```diff
@@ -0,0 +1,90 @@
+[:arrow_left: Back to USKPA
+Documentation](../docs)
+
+### Legacy USKPA database
+
+The legacy data is contained within an MS SQL 2012 database.
+
+
+### Migration
+
+The legacy data migration has 4 primary components, executed in order.
+
+1. Extract data from MS SQL 2012
+2. Prepare extracted data for processing
+3. Transform prepared data to match new data model
+4. Import data to postgres database
+
+
+#### Data Extraction
+
+Selected tables from the legacy database shall be exported to a CSV file.
+
+Tables utilized in migration:
+ - tblCertificate
+ - tblCountries
+ - tblState
+ - tblLicensee
+ - tlbDeliveryStatus
+ - tblHCDCodes
+ - tblPortOfExport
+
+#### Data Preparation
+
+The address fields within the exported `tblCertificates.csv` file contain `\n` characters without surrounding field qualifiers. This corrupts the expected CSV format by having certificate records which span lines leading to truncated and otherwise unexpected data upon import.
+
+To remediate this file, regular expressions are used to identify and replace the offending `\n` characters with `|`. These character replacements will be reverted prior to final import to the new database.
+
+#### Data Transformation
+
+Transformations performed upon incoming Certificate data fields.
+
+##### Status
+  - If no incoming status and `VoidCert==True`, set status to `Certificate.VOID`
+  - If no incoming status and `VoidCert not True`, set status to `Certificate.AVAILABLE`
+  - Otherwise, map incoming status to `Certificate.SHIPPED` or `Certificate.DELIVERED`
+
+##### Licensee
+  - Certificates excluded if `LicenseeID == 17 (TEST)`
+  - Licensee set to `None` if LicenseeID not found in `tblLicensee`
+
+##### AES
+ - Convert `x` to `X`
+ - If value is 14 digits, prepend with `X`.
+ - Imported as-is, even if AES format validation fails.
+
+##### Port of Export
+  - Values modified to be consistent w/ those present in `tblPortOfExport`
+  - Converted to a ForeignKey relationship
+
+##### ImporterCountry
+  - Values modified to conform to latest ISO-3166 short-names.
+
+##### Consignee Address
+  - Insert `ImporterCountry` at end of address, if not already present
+
+#### Data Import
+
+The final data import, and overall orchestration of the migration process is handled via Django management commands.
+
+The data migration should only be executed on a new database which has no `Licensee` or `Certificate` records.
+
+
+Steps to migrate data:
+
+1. Copy exported CSV files into a directory, `./data`, in the same folder as `manage.py`
+2. Load initial data:
+    ```
+    python manage.py loaddata initial_data
+    ```
+3. Import Licensee data:
+    ```
+    python manage.py load_licensees --filepath ./data/tblLicensee.csv
+    ```
+4. Import Certificate data:
+    ```
+    python manage.py load_certs --filepath ./data/tblCertificate.csv
+    ```
+
+Output contains summary information and any warning messages generated by the processing indicating unhandled or otherwise suspicious values.
```

Comment:
Noticed that when migrating certificate information, there were errors flagged at each certificate.  Is that outputted to a local .log file?

Reply:
No, just written to `stdout` at the moment. As-is, It can be directed into a file on execution by the user:
```
docker-compose run app python manage.py load_certs ./data/tblCertificate.csv > output.log
```